ChatServices/IChatServiceConnection.cs

```diff
@@ -11,6 +11,12 @@ public interface IChatServiceConnection
 
         string UserId { get; }
 
+        string UserName { get; }
+
+        string MentionMessageRegExPrefix { get; }
```

Comment:
I'm wondering if there's a way to achieve this without offloading mention parsing to the middleware itself. Like maybe similar to how Mattermost does it, there can be a list of mentions and their locations within the message string? We should discuss.

Reply:
The way to do this would probably be to have the chatServiceConnection parse the mention and strip it before passing on the message to the middleware, but in that case it would strip the mention for every message, not just the summons. I'd be happy to make the change if that's what you'd prefer.

Reply:
It could also annotate it like
```
List<Mention>
{
    uint StartPosition
    uint EndPosition
    string UserId
}
```
and then the middleware could manipulate as it pleases

Reply:
Do you mean having a list of mention metadata objects in addition to the raw message itself? I could be convinced to go down that route, that seems like a decent way to generalize all messages in the data model without specializing mentions too much. Couple questions though:
- Do we expect multiple mentions in a single message?
- Do we want to attempt to support multiple mentions in a single message? (what would be the use case?)
- Would there be merit to annotate the message the other way round? (as in, to always have it stripped but provide metadata like "IsMentioned" so the middleware can reconstruct that if needed)

Reply:
I noticed an "IsMentioned" was already there so I refactored the code to go with my last suggestion and everything looks a bit cleaner to me now.